src/Swagger/DocumentationNormalizer.php

```diff
@@ -163,7 +163,7 @@ public function normalize($object, $format = null, array $context = [])
                         if (null !== $path) {
                             $method = $this->operationMethodResolver->getCollectionOperationMethod($resourceClass, $customOperation);
                             $customSwaggerOperation = $this->getSwaggerOperation($resourceClass, $resourceMetadata, $customOperation, [$method], $prefixedShortName, true, $definitions, $method, $object->getMimeTypes());
-
+                            unset($itemOperationsDocs[$resourceClassIri]);
```

Comment:
I totally agree, I made this WIP for this purpose to improve this part of the code

I will add one.